pydash/strings.py

```diff
@@ -736,6 +736,56 @@ def lines(text):
     return text.splitlines()
 
 
+def lower_case(text):
+    """Converts string to lower case as space separated words. Before
+    converting the given text to lower case this method:
+
+      - replaces special character with white spaces,
+      - adds a white space before the capital letter,
+      - replaces multiple whitespaces with a single whitespace,
+      - pertains spaces between words in the given text,
+      - pertains the digits given in the text.
+
+    Args:
+        text (str): String to convert.
+
+    Returns:
+        str: String converted to lower case as space separated words.
+
+    Example:
+
+        >>> lower_case('fooBar')
+        'foo bar'
+        >>> lower_case('--foo-Bar--')
+        'foo bar'
+        >>> lower_case('/?*Foo10/;"B*Ar')
+        'foo10 b ar'
+    """
+    # TODO: Add release version.
+    characters = []
+
+    for char in text:
+        if not char.isalpha() and not char.isdigit():
+            # If any of the characters is neither an alphabet or a digit, then
+            # replace them with whitespaces.
+            characters.append(' ')
+        elif char.istitle():
+            # If any of the characters is a capital letter, then go ahead and
+            # add a whitespace before the capital letter.
+            characters.append(' {0}'.format(char))
+        else:
+            characters.append(char)
+
+    text = ''.join(characters)
+    text = text.strip()
+
+    # Replace multiple whitespaces between the words in the text with a single
+    # whitespace.
+    text = re.sub("\s\s+", " ", text)
+
+    return text.lower()
```

Comment:
That wouldn't work. There are several cases to deal with. 
1) We need to add whitespaces after and before a digit 
2) We need to replace all the special characters in the text with a whitespaces
3) We need to add space before the capital letter in the text (before lowering the text.

`words()` can't help with all these cases. And also, I would avoid giving complex regex's which would give a lot of false positives or false negatives.